src/Helper/DbArrayHelper.php

```diff
@@ -215,55 +100,62 @@ public static function getValueByPath(object|array $array, Closure|string $key,
      * ]
      * ```
      *
-     * @param array $array The array that needs to be indexed or grouped.
-     * @param string|null $key The column name or anonymous function which result will be used to index the array.
-     * @param array $groups The array of keys that will be used to group the input array by one or more keys. If the
-     * $key attribute or its value for the particular element is null and $groups aren't defined.
-     * The array element will be discarded.
-     * Otherwise, if $groups are specified, an array element will be added to the result array without any key.
+     * @param array[] $array The array that needs to be indexed or arranged.
+     * @param Closure|string|null $indexBy The column name or anonymous function which result will be used to index the
+     * array. If the array does not have the key, the ordinal indexes will be used if `$arrangeBy` is not specified or
+     * a warning will be triggered if `$arrangeBy` is specified.
+     * @param string[] $arrangeBy The array of keys that will be used to arrange the input array by one or more keys.
      *
-     * @throws Exception
-     *
-     * @return array The indexed and/or grouped array.
-     *
-     * @psalm-param array[] $array The array that needs to be indexed or grouped.
-     * @psalm-param string[] $groups The array of keys that will be used to group the input array by one or more keys.
+     * @return array[] The indexed and/or arranged array.
      *
+     * @psalm-param IndexBy|null $indexBy
      * @psalm-suppress MixedArrayAssignment
      */
-    public static function index(array $array, string|null $key = null, array $groups = []): array
+    public static function index(array $array, Closure|string|null $indexBy = null, array $arrangeBy = []): array
     {
+        if (empty($array) || $indexBy === null && empty($arrangeBy)) {
+            return $array;
+        }
+
+        if (empty($arrangeBy)) {
+            if (is_string($indexBy)) {
+                return array_column($array, null, $indexBy);
+            }
+
+            return array_combine(array_map($indexBy, $array), $array);
+        }
+
         $result = [];
 
         foreach ($array as $element) {
             $lastArray = &$result;
 
-            foreach ($groups as $group) {
-                /** @psalm-var string $value */
-                $value = self::getValueByPath($element, $group);
-                if (!array_key_exists($value, $lastArray)) {
+            foreach ($arrangeBy as $group) {
+                $value = (string) $element[$group];
+
+                if (!isset($lastArray[$value])) {
```

Comment:
Can `$lastArray[$value]` be `null`?  If yes, `isset` returns false in this case. It's OK?

Reply:
No, it is an array, empty or non-empty.